src/Filesystem/F.php

```diff
@@ -846,6 +846,18 @@ public static function typeToExtensions(string $type): ?array
 		return static::$types[$type] ?? null;
 	}
 
+	/**
+	 * Safest way of PHP's native unlink function
+	 */
+	public static function unlink(string $file): bool
+	{
+		try {
+			return unlink($file);
+		} catch (Exception $e) {
+			return false;
```

Comment:
Maybe we could check if the file/link is already deleted and treat this case as success. This would catch the race condition error. Also I think we should not swallow the warning by default (to ease debugging e.g. when there are no write permissions or the file system is not available).

```suggestion
			if (file_exists($file) === false) {
				return true;
			}
			
			throw $e;
```

Reply:
I started the PR with this change first but the tests failed. I guess `file_exists` fails on link files. Do you have an idea?

Reply:
Ah, right. `file_exists()` mentions dirs and files, but not links.

Maybe this: `is_file($file) === false && is_link($file) === false`

Reply:
I didn't include it to avoid unnecessary use.  So, do you think there is also a need to use `realpath` before file checks?

Reply:
Oh wait. That doesn't work as `is_file()` resolves links and `is_link()` does not support hardlinks. 🤯

Reply:
Maybe we could check for the "No such file or directory" warning that `unlink()` triggers. But then we need to find a way to handle translated error messages if a translated PHP installation is used.

Reply:
Hm, are translated PHP installations still a thing actually? Can't find any trace of this feature but I'm pretty sure I have seen a German PHP error at some weird hosting provider 10 years ago.

Reply:
I don't think we really need translation of PHP exceptions. It could happen if there were constants defined like in `UPLOAD_*`: https://github.com/getkirby/kirby/blob/3.7.3/src/Api/Api.php#L744-L752

Reply:
What I meant was to check the message of the warning PHP triggers. If the warning message is "No such file or directory", then we know it's fine and we can `return true`. Otherwise, we would throw/trigger that warning again and return `false`.